cmd/modern/main_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package main
+
+import (
+	"errors"
+	"github.com/microsoft/go-sqlcmd/internal/cmdparser"
+	"github.com/microsoft/go-sqlcmd/internal/cmdparser/dependency"
+	"github.com/microsoft/go-sqlcmd/internal/output"
+	"github.com/microsoft/go-sqlcmd/internal/pal"
+	"github.com/microsoft/go-sqlcmd/internal/test"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+func TestMainStart(t *testing.T) {
+	os.Args[1] = "--help"
+	main()
+}
+
+func TestInitializeCallback(t *testing.T) {
+	rootCmd = cmdparser.New[*Root](dependency.Options{})
+	initializeCallback()
+}
+
+func TestDisplayHints(t *testing.T) {
+	buf := test.NewMemoryBuffer()
+	defer buf.Close()
+	outputter = output.New(output.Options{StandardWriter: buf})
+	displayHints([]string{"This is a hint"})
+	assert.Equal(t, pal.LineBreak()+
+		"HINT:"+
+		pal.LineBreak()+
+		"  1. This is a hint"+pal.LineBreak(), buf.String())
+}
+
+func TestCheckErr(t *testing.T) {
+	defer func() { test.CatchExpectedError(recover(), t) }()
```

Comment:
>defer func() { test.CatchExpectedError(recover(), t) }() [](http://example.com/codeflow?start=1&length=56)

does the assert library not have a way to check for a method call leading to a panic?

Reply:
Yes there is, the assert.Panics* functions

https://pkg.go.dev/github.com/stretchr/testify/assert#Panics

I've started a follow-up PR, will replace this pattern with assert.Panics